### PR TITLE
Use checker implementation of isLegalModifier

### PR DESCRIPTION
**Pull request by andy-ms**
We were erroneously excluding some modifiers in `isLegalModifier` in `documentHighlights.ts`. This PR would reuse the existing grammer check from `checker.ts`, which permits more modifiers in smarter ways -- such as `const` in `const enum`.
An alternative would be to scrap `isLegalModifier` and collect all modifiers anyway, relying on diagnostics to tell people when their modifiers are invalid -- I don't really see what the harm in that would be, but it seems to be a well-tested feature so someone must have cared about this.

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -268,6 +268,18 @@ namespace ts {
             },
             getJsxNamespace: () => unescapeLeadingUnderscores(getJsxNamespace()),
             getAccessibleSymbolChain,
+            nodeHasBadModifier: node => {
+                const sourceFile = getSourceFileOfNode(node);
```
Comment (andy-ms) [inline src/compiler/checker.ts]: This is kind of ugly -- `checkGrammarModifiers` could be moved to a pure function returning an optional diagnostic but that have be a big diff.

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -25403,7 +25415,7 @@ namespace ts {
                 }
                 switch (modifier.kind) {
                     case SyntaxKind.ConstKeyword:
-                        if (node.kind !== SyntaxKind.EnumDeclaration && node.parent.kind === SyntaxKind.ClassDeclaration) {
+                        if (node.kind !== SyntaxKind.EnumDeclaration) {
                             return grammarErrorOnNode(node, Diagnostics.A_class_member_cannot_have_the_0_keyword, tokenToString(SyntaxKind.ConstKeyword));
```
Comment (andy-ms) [inline src/compiler/checker.ts]: This error might be misleading on a parameter, should I change it?

Comment (mhegazy) [reply src/compiler/checker.ts]: yeah. makes sense.

**Comment by andy-ms** [comment]
@mhegazy @DanielRosenwasser What would you think of not bothering checking whether the modifier is legal?

**Comment by mhegazy** [comment]
closing in favor of https://github.com/Microsoft/TypeScript/pull/21335

**Pull request closed without merging**